pdl-compiler/src/backends/rust/encoder.rs

```diff
@@ -237,15 +237,47 @@ impl Encoder {
         let shift = self.bit_shift;
 
         match &field.desc {
-            ast::FieldDesc::Flag { optional_field_id, set_value, .. } => {
-                let optional_field_id = optional_field_id.to_ident();
+            ast::FieldDesc::Flag { optional_field_ids, .. } => {
+                let field_id = field.id().unwrap().to_ident();
+                let (cond_field_id, set_value) = &optional_field_ids[0];
+                let cond_field_id_tok = cond_field_id.to_ident();
```

Comment:
nit: `cond_field_id = config_field_id.to_ident()` reusing the same name (that's the pattern everywhere else).